geometric_kernels/spaces/circle.py

```diff
@@ -23,14 +24,14 @@ class SinCosEigenfunctions(EigenfunctionWithAdditionTheorem):
     """
 
     def __init__(self, num_eigenfunctions: int) -> None:
-        assert (
-            num_eigenfunctions % 2 == 1
-        ), "num_eigenfunctions needs to be odd to include all eigenfunctions within a level."
-        assert num_eigenfunctions >= 1
+        # assert (
+        #     num_eigenfunctions % 2 == 1
+        # ), "num_eigenfunctions needs to be odd to include all eigenfunctions within a level."
+        # assert num_eigenfunctions >= 1
 
         self._num_eigenfunctions = num_eigenfunctions
-        # We know `num_eigenfunctions` is odd, therefore:
-        self._num_levels = num_eigenfunctions // 2 + 1
+        # Compute level that fully encompasses all the requested eigenfunctions in a level
+        self._num_levels = ceil(num_eigenfunctions / 2) + 1
```

Comment:
Maybe write a message to std out to inform the user the actual number of eigenfunctions has been updated?